app/api/v2/resource/api_keys.rb

```diff
@@ -47,7 +47,17 @@ def otp_protected!
           declared_params = declared(unified_params, include_missing: false)
                             .except(:totp_code)
                             .merge(scope: params[:scope]&.split(','))
-          api_key = current_user.api_keys.create(declared_params)
+
+          api_key = current_user.api_keys.new(declared_params)
+
+          APIKey.transaction do
+            raise ActiveRecord::Rollback unless api_key.save
```

Comment:
are you sure, that after rollback from this line,   the next check will work?
```
if api_key.errors.any?
	            code_error!(api_key.errors.details, 422)
 end
```

Reply:
yes

Reply:
@dnfd nice!